projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -184,17 +182,17 @@ private[projection] object R2dbcProjectionImpl {
 
     new AdaptedHandler(handlerFactory()) {
       override def process(envelopes: immutable.Seq[Envelope]): Future[Done] = {
-        val acceptedEnvelopes = offsetStore.filterAccepted(envelopes).toVector
-
-        if (acceptedEnvelopes.isEmpty) {
-          FutureDone
-        } else {
-          delegate
-            .process(acceptedEnvelopes)
-            .map { _ =>
-              acceptedEnvelopes.foreach(offsetStore.addInflight)
-              Done
-            }
+        offsetStore.filterAccepted(envelopes).flatMap { acceptedEnvelopes =>
+          if (acceptedEnvelopes.isEmpty) {
+            FutureDone
+          } else {
+            delegate
+              .process(acceptedEnvelopes)
+              .map { _ =>
+                acceptedEnvelopes.foreach(offsetStore.addInflight)
```

Comment:
Not really related to this PR but worth adding a single `addInflight` for multiple envelopes instead of one CAS per envelope?

Reply:
ok, adding to my small fixes todo list